projects/media-viewer/src/lib/annotations/annotation-set/annotation-set.component.ts

```diff
@@ -66,9 +68,10 @@ export class AnnotationSetComponent implements OnInit, OnDestroy {
     this.api
       .postAnnotation(annotation)
       .subscribe(newAnnotation => {
+        const unsavedComment = this.commentService.hasUnsavedComments(annotation);
         const index = this.annotationSet.annotations.findIndex(a => a.id === newAnnotation.id);
 
-        this.annotationSet.annotations[index] = newAnnotation;
+        this.annotationSet.annotations[index] = unsavedComment ? annotation : newAnnotation;
```

Comment:
im abit confused here. from an over view, you are making a post request and then you are checking if the comment is saved or not and from there you change the annotation. Should this be here?